src/ctap/storage.rs

```diff
@@ -59,11 +62,24 @@ const PIN_RETRIES: usize = 4;
 const ATTESTATION_PRIVATE_KEY: usize = 5;
 const ATTESTATION_CERTIFICATE: usize = 6;
 const AAGUID: usize = 7;
-const NUM_TAGS: usize = 8;
+#[cfg(feature = "with_ctap2_1")]
+const MIN_PIN_LENGTH: usize = 8;
+#[cfg(feature = "with_ctap2_1")]
+const MIN_PIN_LENGTH_RP_IDS: usize = 9;
+// Different NUM_TAGS depending on the CTAP version make the storage incompatible,
+// so we use the maximum.
+const NUM_TAGS: usize = 10;
 
 const MAX_PIN_RETRIES: u8 = 6;
 const ATTESTATION_PRIVATE_KEY_LENGTH: usize = 32;
 const AAGUID_LENGTH: usize = 16;
+#[cfg(feature = "with_ctap2_1")]
+const DEFAULT_MIN_PIN_LENGTH: u8 = 4;
+#[cfg(feature = "with_ctap2_1")]
+const _DEFAULT_MIN_PIN_LENGTH_RP_IDS: Vec<String> = Vec::new();
```

Comment:
Any reason to prefix this with an underscore? If it's unused, shouldn't it be `#[cfg(test)]` instead?

Reply:
The reason there is an unused variable is because there is the minPinLength extension that is spread between comments. In this case, the clientPin command contains instructions about storing relying parties. This is not conditioned on the extension actually existing, but storing the data is only useful if you really support the extension. But, those extra flash writes are unncessary, so I disabled them for until the extension is implemented. The new clientPin command is implemented as described, usage is postponed to a later PR that includes the extension.

There is no clear separation between extension and protocol in the specification. This feature is not really only for testing, the test exists for the actual feature that just doesn't make sense without the extension to read the data. I could also move all that unused code to a later PR, and document the gap in the implementation?

Reply:
I have the commits for the extension ready, let me know if you want them to be in a separate PR or not.

Reply:
Let's do a separate PR once this one is merged.